### ISSUE TITLE
RAC DatePicker: Too many re-renders error 

### ISSUE THREAD

**Issue by ilsanchez** [adobe/react-spectrum#5208]
### Provide a general summary of the issue here

Hi!

First of all for this fantastic library, I'm really enjoying using it.

There seems to be a bug in the DatePicker or Calendar component (not sure which of the two may be the cause). It seems that when the value of the component is invalid, at some point, it enters in a render loop.

### 🤔 Expected Behavior?

We would need to be able to use the value of a DatePicker as maxValue or minValue of another DatePicker to ensure, for example, that the start date is not higher than the end date when selected via the calendar

### 😯 Current Behavior

When using two DatePickers linked to each other through the minValue and maxValue properties, if the value of one of the DatePickers is incorrect, when opening the calendar it gives an error `Too many re-renders. React limits the number of renders to prevent an infinite loop.`

### 🔦 Context

The attached example I think is a real use case that we can find very often on the web.

### 🖥️ Steps to Reproduce

CodeSandBox: https://codesandbox.io/s/vigilant-shannon-2khs5l

Steps: 
 - Set start date month on february writing in the input (12-2-2023)
 - Set end date month on january writing in the input (24-10-2023)
 - Both DatePicker should be invalid
 - Open start date calendar

### Version

1.0.0-beta.1 (also tried on 1.0.0-beta.0)

### What browsers are you seeing the problem on?

Firefox, Chrome

### What operating system are you using?

Ubuntu 20.04.6 LTS

**Comment by yihuiliao** [comment]
I’m glad to hear that you’re enjoying the library! I took an initial glance at your codesandbox with the steps you gave to reproduce and if I set the values to what you specified, I get the error if I open the first DatePicker but not the second DatePicker even though technically both DatePickers are invalid. This seems to be because at least for the first DatePicker, the maxValue is actually less than the minValue.

In the case that the maxValue is somehow able to be less than the minValue, I’m not really sure what you would expect since the range is already kinda weird. 

I guess what we could do if this were to happen is not allow the calendar popover to open? So even if you tried clicking it, nothing would happen. This would probably need a bit more work/thinking to determine what type of behaviors we would expect in this scenario though.

**Comment by yihuiliao** [comment]
Of course! I think throwing an error sounds very reasonable in the case this should ever happen. Thanks for your suggestion!

**Comment by reidbarber** [comment]
We would eventually like to show some warnings/errors only in development environments, so this may be a good candidate for that: https://github.com/adobe/react-spectrum/issues/2944

**Comment by devongovett** [comment]
No longer reproduces.
